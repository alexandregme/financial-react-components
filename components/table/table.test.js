import React from 'react';
import { shallow, mount } from 'enzyme';
import Table from 'components/table';

describe('Table', () => {
  let mountedTable;
  let table;

  beforeEach(() => {
    mountedTable = shallow(<Table />);
    table = mountedTable.find('table');
  });

  it('renders without crashing', () => {
    shallow(<Table />);
    expect(mountedTable).toMatchSnapshot();
  });

  it('contains a form element', () => {
    expect(table).toHaveLength(1);
  });

  it('expect label have table table wrapper', () => {
    const tableWrapper = mountedTable.find('div.tablewrap');
    expect(tableWrapper).toHaveLength(1);
  });

  it('expect label have children component with a span tag', () => {
    expect(table.find('thead')).toHaveLength(1);
  });

  it('expect basic definition for styling', () => {
    expect(table.props().className).toEqual('table table-striped table-hover');
  });
});

describe('Table custom props', () => {
  let mountedTable;
  let table;

  const props = {
    children: <tbody />,
    className: 'customClassName',
  };

  beforeEach(() => {
    mountedTable = mount(<Table {...props} />);
    table = mountedTable.find('table');
  });

  it('expect table have custom children', () => {
    expect(mountedTable.props().children).toEqual(props.children);
  });

  it('expect basic definition for styling', () => {
    expect(table.props().className).toEqual('table table-striped table-hover customClassName');
  });
});
