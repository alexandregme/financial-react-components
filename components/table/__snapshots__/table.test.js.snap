// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders without crashing 1`] = `
<div
  className="tablewrap"
>
  <table
    className="table table-striped table-hover"
  >
    <thead />
  </table>
</div>
`;
